src/Features/Core/Portable/SemanticSearch/QueryExecutionContext.cs

```diff
@@ -130,30 +130,49 @@ private async ValueTask InvokeAsync(Project project, Compilation compilation, ob
 
             try
             {
-                var symbols = (IEnumerable<ISymbol?>?)method.Invoke(null, [entity]) ?? [];
+                var symbols = method.Invoke(null, [entity]);
+                if (symbols != null)
+                {
+                    if (isAsync)
+                    {
+                        await foreach (var symbol in ((IAsyncEnumerable<ISymbol?>)symbols).WithCancellation(cancellationToken).ConfigureAwait(false))
+                        {
+                            await ObserveSymbolAsync(symbol).ConfigureAwait(false);
+                        }
+                    }
+                    else
+                    {
+                        foreach (var symbol in (IEnumerable<ISymbol?>)symbols)
+                        {
+                            await ObserveSymbolAsync(symbol).ConfigureAwait(false);
+                        }
+                    }
+                }
 
-                foreach (var symbol in symbols)
+                async ValueTask ObserveSymbolAsync(ISymbol? symbol)
                 {
                     cancellationToken.ThrowIfCancellationRequested();
 
-                    if (symbol != null)
+                    if (symbol == null)
                     {
-                        executionTime += Stopwatch.GetElapsedTime(executionStart);
+                        return;
+                    }
 
-                        try
-                        {
-                            var definitionItem = await symbol.ToClassifiedDefinitionItemAsync(
-                                classificationOptions, project.Solution, s_findReferencesSearchOptions, isPrimary: true, includeHiddenLocations: false, cancellationToken).ConfigureAwait(false);
+                    executionTime += Stopwatch.GetElapsedTime(executionStart);
```

Comment:
aside: odd that all the symbols are getting "isPrimary: true".  but probably doesn't matter that much.